### PR TITLE
Analyses v2 Fixes and Additions

### PR DESCRIPTION
**Pull request by PerryShafran-NOAA**
This PR addresses many of the items from the [Fixes and Additions](https://docs.google.com/document/d/1vOGnCwCM3COUhQlTqHyBZrYSCPx9b6AYnxLc5NtVGEk/edit) document for Analyses for EVSv2, including the following:

- Changing subprocess.Popen to subprocess.run.  This also included fixing a second line to the routine. 
- Define $NET prior to $HOMEevs in J-job
- Removing concatenation of METplus logs in the stats code
- Removing LOOP ORDER from parm files
- Removing #PBS -V option from all ecf and driver scripts
- No warnings when using METplus v6.0.0 beta

To run these items, clone my fork, and checkout the feature/analyses_v2 branch.  Be sure you ln to the fix directory.

For grid2obs stats, run each job using qsub -v vhr=$vhr, where $vhr=00,01,02.....,23.  It is not necessary to run every single hour, but it is beneficial to at least run for vhr=00 and vhr=01, and maybe vhr=23 to test the gather step.  For the precip stats, run each job using vhr=12, 13, and 23.  

For plots, run the plot job (no vhr=$vhr is needed for the plot step).  

Complete PR by 8/19/24.

### CODE REVIEW

File: jobs/JEVS_ANALYSES_PLOTS
Comment (malloryprow) [inline jobs/JEVS_ANALYSES_PLOTS:1]: It looks like the export statements of NET, STEP, COMPONENT, and RUN are all commented out.

File: jobs/JEVS_ANALYSES_STATS
Comment (malloryprow) [inline jobs/JEVS_ANALYSES_STATS:1]: It looks like the export statements of NET, STEP, COMPONENT, and RUN are all commented out.

File: jobs/JEVS_ANALYSES_PLOTS
Comment (PerryShafran-NOAA) [reply jobs/JEVS_ANALYSES_PLOTS:1]: Fixed.

File: jobs/JEVS_ANALYSES_STATS
Comment (PerryShafran-NOAA) [reply jobs/JEVS_ANALYSES_STATS:1]: Fixed.

**Comment by malloryprow** [comment]
Starting the stats step testing with the precip jobs

**COMOUT:**/lfs/h2/emc/vpppg/noscrub/mallory.row/verification/EVS_PRs/pr512/evs/v2.0

1. jevs_analyses_ccpa_precip_stats.sh with vhr=00

> **Log File:** /lfs/h2/emc/vpppg/noscrub/mallory.row/verification/EVS_PRs/pr512/EVS/dev/drivers/scripts/stats/analyses/jevs_analyses_ccpa_precip_stats.o148123196
> **DATA:** /lfs/h2/emc/stmp/mallory.row/evs_test/prod/tmp/jevs_analyses_ccpa_precip_stats.148123196.cbqs01

2. jevs_analyses_ccpa_precip_stats.sh with vhr=01
> **Log File:** /lfs/h2/emc/vpppg/noscrub/mallory.row/verification/EVS_PRs/pr512/EVS/dev/drivers/scripts/stats/analyses/jevs_analyses_ccpa_precip_stats.o148123743
> **DATA:** /lfs/h2/emc/stmp/mallory.row/evs_test/prod/tmp/jevs_analyses_ccpa_precip_stats.148123743.cbqs01

3. jevs_analyses_ccpa_precip_stats.sh with vhr=23
> **Log File:** /lfs/h2/emc/vpppg/noscrub/mallory.row/verification/EVS_PRs/pr512/EVS/dev/drivers/scripts/stats/analyses/jevs_analyses_ccpa_precip_stats.o148124010
> **DATA:** /lfs/h2/emc/stmp/mallory.row/evs_test/prod/tmp/jevs_analyses_ccpa_precip_stats.148124010.cbqs01

4. jevs_analyses_rtma_precip_stats.sh with vhr=00
> **Log File:** /lfs/h2/emc/vpppg/noscrub/mallory.row/verification/EVS_PRs/pr512/EVS/dev/drivers/scripts/stats/analyses/jevs_analyses_rtma_precip_stats.o148125347
> **DATA:** /lfs/h2/emc/stmp/mallory.row/evs_test/prod/tmp/jevs_analyses_rtma_precip_stats.148125347.cbqs01

5. jevs_analyses_rtma_precip_stats.sh with vhr=01
> **Log File:** /lfs/h2/emc/vpppg/noscrub/mallory.row/verification/EVS_PRs/pr512/EVS/dev/drivers/scripts/stats/analyses/jevs_analyses_rtma_precip_stats.o148126076
> **DATA:** /lfs/h2/emc/stmp/mallory.row/evs_test/prod/tmp/jevs_analyses_rtma_precip_stats.148126076.cbqs01

6. jevs_analyses_rtma_precip_stats.sh with vhr=23
> **Log File:** /lfs/h2/emc/vpppg/noscrub/mallory.row/verification/EVS_PRs/pr512/EVS/dev/drivers/scripts/stats/analyses/jevs_analyses_rtma_precip_stats.o148126667
> **DATA:** /lfs/h2/emc/stmp/mallory.row/evs_test/prod/tmp/jevs_analyses_rtma_precip_stats.148126667.cbqs01

7. jevs_analyses_urma_precip_stats.sh with vhr=00
> **Log File:** /lfs/h2/emc/vpppg/noscrub/mallory.row/verification/EVS_PRs/pr512/EVS/dev/drivers/scripts/stats/analyses/jevs_analyses_urma_precip_stats.o148127688
> **DATA:** /lfs/h2/emc/stmp/mallory.row/evs_test/prod/tmp/jevs_analyses_urma_precip_stats.148127688.cbqs01

8. jevs_analyses_urma_precip_stats.sh with vhr=01
> **Log File:** /lfs/h2/emc/vpppg/noscrub/mallory.row/verification/EVS_PRs/pr512/EVS/dev/drivers/scripts/stats/analyses/jevs_analyses_urma_precip_stats.o148128175
> **DATA:** /lfs/h2/emc/stmp/mallory.row/evs_test/prod/tmp/jevs_analyses_urma_precip_stats.148128175.cbqs01

9. jevs_analyses_urma_precip_stats.sh with vhr=23
> **Log File:** /lfs/h2/emc/vpppg/noscrub/mallory.row/verification/EVS_PRs/pr512/EVS/dev/drivers/scripts/stats/analyses/jevs_analyses_urma_precip_stats.o148128421
> **DATA:** /lfs/h2/emc/stmp/mallory.row/evs_test/prod/tmp/jevs_analyses_urma_precip_stats.148128421.cbqs01

**Comment by PerryShafran-NOAA** [comment]
@malloryprow For precip I should have said run with vhr=12 and vhr=13 as you get better results at those times.  I'm sorry, I'll edit the instructions.

**Comment by malloryprow** [comment]
**COMOUT:**/lfs/h2/emc/vpppg/noscrub/mallory.row/verification/EVS_PRs/pr512/evs/v2.0

1. jevs_analyses_ccpa_precip_stats.sh with vhr=12

> **Log File:** /lfs/h2/emc/vpppg/noscrub/mallory.row/verification/EVS_PRs/pr512/EVS/dev/drivers/scripts/stats/analyses/jevs_analyses_ccpa_precip_stats.o148130963
> **DATA:** /lfs/h2/emc/stmp/mallory.row/evs_test/prod/tmp/jevs_analyses_ccpa_precip_stats.148130963.cbqs01

2. jevs_analyses_ccpa_precip_stats.sh with vhr=13
> **Log File:** /lfs/h2/emc/vpppg/noscrub/mallory.row/verification/EVS_PRs/pr512/EVS/dev/drivers/scripts/stats/analyses/jevs_analyses_ccpa_precip_stats.o148131405
> **DATA:** /lfs/h2/emc/stmp/mallory.row/evs_test/prod/tmp/jevs_analyses_ccpa_precip_stats.148131405.cbqs01

3. jevs_analyses_ccpa_precip_stats.sh with vhr=23
> **Log File:** /lfs/h2/emc/vpppg/noscrub/mallory.row/verification/EVS_PRs/pr512/EVS/dev/drivers/scripts/stats/analyses/jevs_analyses_ccpa_precip_stats.o148132200
> **DATA:** /lfs/h2/emc/stmp/mallory.row/evs_test/prod/tmp/jevs_analyses_ccpa_precip_stats.148132200.cbqs01

4. jevs_analyses_rtma_precip_stats.sh with vhr=12
> **Log File:** /lfs/h2/emc/vpppg/noscrub/mallory.row/verification/EVS_PRs/pr512/EVS/dev/drivers/scripts/stats/analyses/jevs_analyses_rtma_precip_stats.o148131574
> **DATA:** /lfs/h2/emc/stmp/mallory.row/evs_test/prod/tmp/jevs_analyses_rtma_precip_stats.148131574.cbqs01

5. jevs_analyses_rtma_precip_stats.sh with vhr=13
> **Log File:** /lfs/h2/emc/vpppg/noscrub/mallory.row/verification/EVS_PRs/pr512/EVS/dev/drivers/scripts/stats/analyses/jevs_analyses_rtma_precip_stats.o148131576
> **DATA:** /lfs/h2/emc/stmp/mallory.row/evs_test/prod/tmp/jevs_analyses_rtma_precip_stats.148131576.cbqs01

6. jevs_analyses_rtma_precip_stats.sh with vhr=23
> **Log File:** /lfs/h2/emc/vpppg/noscrub/mallory.row/verification/EVS_PRs/pr512/EVS/dev/drivers/scripts/stats/analyses/jevs_analyses_rtma_precip_stats.o148132370
> **DATA:** /lfs/h2/emc/stmp/mallory.row/evs_test/prod/tmp/jevs_analyses_rtma_precip_stats.148132370.cbqs01

7. jevs_analyses_urma_precip_stats.sh with vhr=12
> **Log File:** /lfs/h2/emc/vpppg/noscrub/mallory.row/verification/EVS_PRs/pr512/EVS/dev/drivers/scripts/stats/analyses/jevs_analyses_urma_precip_stats.o148131736
> **DATA:** /lfs/h2/emc/stmp/mallory.row/evs_test/prod/tmp/jevs_analyses_urma_precip_stats.148131736.cbqs01

8. jevs_analyses_urma_precip_stats.sh with vhr=13
> **Log File:** /lfs/h2/emc/vpppg/noscrub/mallory.row/verification/EVS_PRs/pr512/EVS/dev/drivers/scripts/stats/analyses/jevs_analyses_urma_precip_stats.o148131821
> **DATA:** /lfs/h2/emc/stmp/mallory.row/evs_test/prod/tmp/jevs_analyses_urma_precip_stats.148131821.cbqs01

9. jevs_analyses_urma_precip_stats.sh with vhr=23
> **Log File:** /lfs/h2/emc/vpppg/noscrub/mallory.row/verification/EVS_PRs/pr512/EVS/dev/drivers/scripts/stats/analyses/jevs_analyses_urma_precip_stats.o148132577
> **DATA:** /lfs/h2/emc/stmp/mallory.row/evs_test/prod/tmp/jevs_analyses_urma_precip_stats.148132577.cbqs01

**Comment by PerryShafran-NOAA** [comment]
All the precip analyses are identical to emc.vpppg.  There are no ERRORs or WARNINGs in any of the .o files.

**Comment by malloryprow** [comment]
grid2obs stats jobs

**COMOUT:**/lfs/h2/emc/vpppg/noscrub/mallory.row/verification/EVS_PRs/pr512/evs/v2.0

1.jevs_analyses_rtma_grid2obs_stats.sh with vhr=00
> **Log File:** /lfs/h2/emc/vpppg/noscrub/mallory.row/verification/EVS_PRs/pr512/EVS/dev/drivers/scripts/stats/analyses/jevs_analyses_rtma_grid2obs_stats.o148134040
> **DATA:** /lfs/h2/emc/stmp/mallory.row/evs_test/prod/tmp/jevs_analyses_rtma_grid2obs_stats.148134040.cbqs01

2.jevs_analyses_rtma_grid2obs_stats.sh with vhr=01
> **Log File:** /lfs/h2/emc/vpppg/noscrub/mallory.row/verification/EVS_PRs/pr512/EVS/dev/drivers/scripts/stats/analyses/jevs_analyses_rtma_grid2obs_stats.o148134289
> **DATA:** /lfs/h2/emc/stmp/mallory.row/evs_test/prod/tmp/jevs_analyses_rtma_grid2obs_stats.148134289.cbqs01

3.jevs_analyses_rtma_grid2obs_stats.sh with vhr=23
> **Log File:** /lfs/h2/emc/vpppg/noscrub/mallory.row/verification/EVS_PRs/pr512/EVS/dev/drivers/scripts/stats/analyses/jevs_analyses_rtma_grid2obs_stats.o148138785
> **DATA:** /lfs/h2/emc/stmp/mallory.row/evs_test/prod/tmp/jevs_analyses_rtma_grid2obs_stats.148138785.cbqs01

4.jevs_analyses_rtma_ru_grid2obs_stats.sh with vhr=00
> **Log File:** /lfs/h2/emc/vpppg/noscrub/mallory.row/verification/EVS_PRs/pr512/EVS/dev/drivers/scripts/stats/analyses/jevs_analyses_rtma_ru_grid2obs_stats.o148139439
> **DATA:** /lfs/h2/emc/stmp/mallory.row/evs_test/prod/tmp/jevs_analyses_rtma_ru_grid2obs_stats.148139439.cbqs01

5.jevs_analyses_rtma_ru_grid2obs_stats.sh with vhr=01
> **Log File:** /lfs/h2/emc/vpppg/noscrub/mallory.row/verification/EVS_PRs/pr512/EVS/dev/drivers/scripts/stats/analyses/jevs_analyses_rtma_ru_grid2obs_stats.o148139454
> **DATA:** /lfs/h2/emc/stmp/mallory.row/evs_test/prod/tmp/jevs_analyses_rtma_ru_grid2obs_stats.148139454.cbqs01

6.jevs_analyses_rtma_ru_grid2obs_stats.sh with vhr=23
> **Log File:** /lfs/h2/emc/vpppg/noscrub/mallory.row/verification/EVS_PRs/pr512/EVS/dev/drivers/scripts/stats/analyses/jevs_analyses_rtma_ru_grid2obs_stats.o148140945
> **DATA:** /lfs/h2/emc/stmp/mallory.row/evs_test/prod/tmp/jevs_analyses_rtma_ru_grid2obs_stats.148140945.cbqs01

7.jevs_analyses_urma_grid2obs_stats.sh with vhr=00
> **Log File:** /lfs/h2/emc/vpppg/noscrub/mallory.row/verification/EVS_PRs/pr512/EVS/dev/drivers/scripts/stats/analyses/jevs_analyses_urma_grid2obs_stats.o148140346
> **DATA:** /lfs/h2/emc/stmp/mallory.row/evs_test/prod/tmp/jevs_analyses_urma_grid2obs_stats.148140346.cbqs01

8.jevs_analyses_urma_grid2obs_stats.sh with vhr=01
> **Log File:** /lfs/h2/emc/vpppg/noscrub/mallory.row/verification/EVS_PRs/pr512/EVS/dev/drivers/scripts/stats/analyses/jevs_analyses_urma_grid2obs_stats.o148140372
> **DATA:** /lfs/h2/emc/stmp/mallory.row/evs_test/prod/tmp/jevs_analyses_urma_grid2obs_stats.148140372.cbqs01

9.jevs_analyses_urma_grid2obs_stats.sh with vhr=23 (still running at the time of this comment)
> **Log File:** /lfs/h2/emc/vpppg/noscrub/mallory.row/verification/EVS_PRs/pr512/EVS/dev/drivers/scripts/stats/analyses/jevs_analyses_urma_grid2obs_stats.o148145143
> **DATA:** /lfs/h2/emc/stmp/mallory.row/evs_test/prod/tmp/jevs_analyses_urma_grid2obs_stats.148145143.cbqs01

**Comment by PerryShafran-NOAA** [comment]
The grid2obs stats are identical with emc.vpppg, and there are no ERRORs or WARNINGs in the .o files.

**Comment by malloryprow** [comment]
The URMA grid2obs vhr=23 stats are still running. Can you check that when it is done? We can move on to plots tomorrow if that job is good too.

**Comment by PerryShafran-NOAA** [comment]
OK, we can do that.

**Comment by malloryprow** [comment]
I noticed a few things in the drivers too that aren't related to the changes of the PR

1. Commented out lines ecflow lines
2. The purpose at the bottom of the script does not reflect what the driver script is doing

**Comment by PerryShafran-NOAA** [comment]
I can make a few edits in the meantime, if that's helpful.

**Comment by malloryprow** [comment]
Running the plots and pointing COMIN to /lfs/h2/emc/vpppg/noscrub/emc.vpppg/evs/v2.0.

jevs_analyses_grid2obs_plots.sh
**Log File:** /lfs/h2/emc/vpppg/noscrub/mallory.row/verification/EVS_PRs/pr512/EVS/dev/drivers/scripts/plots/analyses/jevs_analyses_grid2obs_plots.o148268563
**COMOUT:** /lfs/h2/emc/vpppg/noscrub/mallory.row/verification/EVS_PRs/pr512/evs/v2.0
**DATA:** /lfs/h2/emc/stmp/mallory.row/evs_test/prod/tmp/jevs_analyses_grid2obs_plots.148268563.cbqs01

The purpose in the plots driver is also incorrect.

**Comment by PerryShafran-NOAA** [comment]
@malloryprow The vhr=23 stats output look good.

The plot job is complete and looks identical to the emc.vpppg.  There are no ERRORs.  There is a WARNING for a missing stats file, but that WARNING is in the emc.vpppg run and is OK if there is a missing stats file.  Therefore, I would be OK with validating the plot job.

**Review by malloryprow — APPROVED** [review]

**Comment by malloryprow** [comment]
@AndrewBenjamin-NOAA Is it sorted out you can approve PRs now?

**Comment by AndrewBenjamin-NOAA** [comment]
> @AndrewBenjamin-NOAA Is it sorted out you can approve PRs now?

Yes, I believe so, I'll start my review now.

**Review by AndrewBenjamin-NOAA — APPROVED** [review]
These look good, I approve these changes.

**Pull request merged**